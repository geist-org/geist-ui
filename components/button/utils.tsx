import React, { ReactNode } from 'react'
import ButtonIcon from './button-icon'
import { ButtonProps } from './button'
import { ButtonGroupConfig } from '../button-group/button-group-context'
import useClasses from '../use-classes'

export const getButtonChildrenWithIcon = (
  auto: boolean,
  children: ReactNode,
  icons: {
    icon?: React.ReactNode
    iconRight?: React.ReactNode
  },
) => {
  const { icon, iconRight } = icons
  const hasIcon = icon || iconRight
  const isRight = Boolean(iconRight)
  const paddingForAutoMode = auto
    ? `calc(var(--geist-ui-button-height) / 2 + var(--geist-ui-button-icon-padding) * .5)`
    : 0
  const classes = useClasses('text', isRight ? 'right' : 'left')

  if (!hasIcon) return <div className="text">{children}</div>
  if (React.Children.count(children) === 0) {
    return (
      <ButtonIcon isRight={isRight} isSingle>
        {hasIcon}
      </ButtonIcon>
    )
  }
  return (
    <>
      <ButtonIcon isRight={isRight}>{hasIcon}</ButtonIcon>
      <div className={classes}>
        {children}
        <style jsx>{`
          .left {
            padding-left: ${paddingForAutoMode};
          }
          .right {
            padding-right: ${paddingForAutoMode};
          }
        `}</style>
      </div>
    </>
  )
}

export const filterPropsWithGroup = <T extends React.PropsWithChildren<ButtonProps>>(
  props: T,
  config: ButtonGroupConfig,
): T => {
  if (!config.isButtonGroup) return props
  return {
    ...props,
    auto: true,
    shadow: false,
    ghost: config.ghost || props.ghost,
    type: config.type || props.type,
    disabled: config.disabled || props.disabled,
  }
}
