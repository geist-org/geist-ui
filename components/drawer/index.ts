import Drawer from './drawer'
import ModalTitle from '../modal/modal-title'
import ModalSubtitle from '../modal/modal-subtitle'
import ModalContent from '../modal/modal-content'

export type DrawerComponentType = typeof Drawer & {
  Title: typeof ModalTitle
  Subtitle: typeof ModalSubtitle
  Content: typeof ModalContent
}
;(Drawer as DrawerComponentType).Title = ModalTitle
;(Drawer as DrawerComponentType).Subtitle = ModalSubtitle
;(Drawer as DrawerComponentType).Content = ModalContent

export type { DrawerProps } from './drawer'
export type {
  ModalTitleProps as DrawerTitleProps,
  ModalSubtitleProps as DrawerSubtitleProps,
  ModalContentProps as DrawerContentProps,
} from '../modal'
export default Drawer as DrawerComponentType
