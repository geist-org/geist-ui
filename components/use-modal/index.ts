import useModal from './use-modal'

export type { ModalHooksBindings } from './use-modal'
export default useModal
