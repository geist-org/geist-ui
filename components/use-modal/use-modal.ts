import { Dispatch, MutableRefObject, SetStateAction } from 'react'
import useCurrentState from '../utils/use-current-state'
import { ModalProps } from '../modal'

export type ModalHooksBindings = Pick<ModalProps, 'visible' | 'onClose'>

const useModal = (
  initialVisible: boolean = false,
): {
  visible: boolean
  setVisible: Dispatch<SetStateAction<boolean>>
  currentRef: MutableRefObject<boolean>
  bindings: ModalHooksBindings
} => {
  const [visible, setVisible, currentRef] = useCurrentState<boolean>(initialVisible)

  return {
    visible,
    setVisible,
    currentRef,
    bindings: {
      visible,
      onClose: () => setVisible(false),
    },
  }
}

export default useModal
