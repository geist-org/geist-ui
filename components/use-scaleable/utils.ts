import { ScaleableProps } from './scaleable-context'

export type ScaleablePropsAndInvalid = keyof ScaleableProps | 'size'

export const ScaleablePropKeys: Array<ScaleablePropsAndInvalid> = [
  'paddingLeft',
  'pl',
  'paddingRight',
  'pr',
  'paddingTop',
  'pt',
  'paddingBottom',
  'pb',
  'marginTop',
  'mt',
  'marginRight',
  'mr',
  'marginBottom',
  'mb',
  'marginLeft',
  'ml',
  'px',
  'py',
  'mx',
  'my',
  'width',
  'height',
  'font',
  'unit',
  'scale',
  'size',
]

export const filterScaleableProps = <T extends Record<any, any>>(props: T) => {
  const keys = Object.keys(props).filter(key => key !== '')
  const nextProps: any = {}
  for (const key of keys) {
    if (!(ScaleablePropKeys as string[]).includes(key)) {
      nextProps[key] = props[key]
    }
  }
  return nextProps as Omit<T, ScaleablePropsAndInvalid>
}
