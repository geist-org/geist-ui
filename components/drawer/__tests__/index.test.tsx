import React from 'react'
import { mount } from 'enzyme'
import { Drawer } from 'components'
import { nativeEvent, updateWrapper } from 'tests/utils'
import { expectDrawerIsClosed, expectDrawerIsOpened } from './use-modal.test'
import userEvent from '@testing-library/user-event'

describe('Drawer', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Drawer visible={true}>
        <Drawer.Title>Drawer</Drawer.Title>
        <Drawer.Subtitle>This is a drawer</Drawer.Subtitle>
        <Drawer.Content>
          <p>Some content contained within the drawer.</p>
        </Drawer.Content>
      </Drawer>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should work correctly with different placement', () => {
    const top = mount(
      <Drawer visible={true} placement="top">
        <p>Some content contained within the drawer.</p>
      </Drawer>,
    )
    expect(top.html()).toMatchSnapshot()
    expect(() => top.unmount()).not.toThrow()

    const right = mount(
      <Drawer visible={true} placement="right">
        <p>Some content contained within the drawer.</p>
      </Drawer>,
    )
    expect(right.html()).toMatchSnapshot()
    expect(() => right.unmount()).not.toThrow()

    const bottom = mount(
      <Drawer visible={true} placement="bottom">
        <p>Some content contained within the drawer.</p>
      </Drawer>,
    )
    expect(bottom.html()).toMatchSnapshot()
    expect(() => bottom.unmount()).not.toThrow()

    const left = mount(
      <Drawer visible={true} placement="left">
        <p>Some content contained within the drawer.</p>
      </Drawer>,
    )
    expect(left.html()).toMatchSnapshot()
    expect(() => left.unmount()).not.toThrow()
  })

  it('should trigger event when drawer changed', async () => {
    const closeHandler = jest.fn()
    const wrapper = mount(
      <Drawer onClose={closeHandler}>
        <Drawer.Title>Modal</Drawer.Title>
      </Drawer>,
    )
    expectDrawerIsClosed(wrapper)

    wrapper.setProps({ visible: true })
    await updateWrapper(wrapper, 350)
    expectDrawerIsOpened(wrapper)

    wrapper.find('.backdrop').simulate('click', nativeEvent)
    await updateWrapper(wrapper, 500)
    expectDrawerIsClosed(wrapper)
    expect(closeHandler).toHaveBeenCalled()
  })

  it('should disable backdrop event', async () => {
    const closeHandler = jest.fn()
    const wrapper = mount(
      <Drawer visible={true} disableBackdropClick onClose={closeHandler}>
        <Drawer.Title>Modal</Drawer.Title>
      </Drawer>,
    )
    wrapper.find('.backdrop').simulate('click', nativeEvent)
    await updateWrapper(wrapper, 500)
    expectDrawerIsOpened(wrapper)
    expect(closeHandler).not.toHaveBeenCalled()
  })

  it('customization should be supported', () => {
    const wrapper = mount(
      <Drawer visible={true} width="100px" wrapClassName="test-class">
        <Drawer.Title>Modal</Drawer.Title>
      </Drawer>,
    )
    const html = wrapper.find('.wrapper').html()
    expect(html).toMatchSnapshot()
    expect(wrapper.find('.wrapper').at(0).getDOMNode()).toHaveClass('test-class')
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('focus should only be switched within modal', async () => {
    const wrapper = mount(
      <Drawer visible={true} width="100px" wrapClassName="test-class">
        <button id="button" />
      </Drawer>,
    )
    const tabStart = wrapper.find('.hide-tab').at(0).getDOMNode()
    const tabEnd = wrapper.find('.hide-tab').at(1).getDOMNode()
    const button = wrapper.find('#button').at(0).getDOMNode()
    const focusTrap = wrapper.find('.wrapper').at(0).getDOMNode()

    expect(tabStart).toHaveFocus()
    userEvent.tab({ focusTrap })
    expect(button).toHaveFocus()
    userEvent.tab()
    expect(tabEnd).toHaveFocus()
    userEvent.tab()
    expect(tabStart).toHaveFocus()

    userEvent.tab({ shift: true, focusTrap })
    expect(tabEnd).toHaveFocus()
    userEvent.tab({ shift: true, focusTrap })
    expect(button).toHaveFocus()
    userEvent.tab({ shift: true, focusTrap })
    expect(tabStart).toHaveFocus()
  })

  it('should close drawer when keyboard event is triggered', async () => {
    const wrapper = mount(
      <Drawer visible={true}>
        <Drawer.Title>Drawer</Drawer.Title>
      </Drawer>,
    )
    expectDrawerIsOpened(wrapper)
    userEvent.keyboard('{esc}')
    await updateWrapper(wrapper, 500)
    expectDrawerIsClosed(wrapper)
  })

  it('should prevent close modal when keyboard is false', async () => {
    const wrapper = mount(
      <Drawer visible={true} keyboard={false}>
        <Drawer.Title>Drawer</Drawer.Title>
      </Drawer>,
    )
    expectDrawerIsOpened(wrapper)
    userEvent.keyboard('{esc}')
    await updateWrapper(wrapper, 500)
    expectDrawerIsOpened(wrapper)
  })
})
