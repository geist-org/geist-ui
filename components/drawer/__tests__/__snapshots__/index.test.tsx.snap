// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer customization should be supported 1`] = `
"<div class=\\"wrapper right test-class  wrapper-enter\\" role=\\"dialog\\" tabindex=\\"-1\\"><div tabindex=\\"0\\" class=\\"hide-tab start\\" aria-hidden=\\"true\\"></div><h2 class=\\"\\">Modal</h2><style>
        h2 {
          line-height: 1.6;
          font-weight: normal;
          text-align: center;
          display: inline-flex;
          flex-shrink: 0;
          justify-content: center;
          align-items: center;
          word-break: break-word;
          text-transform: capitalize;
          font-size: calc(1.5 * 16px);
          color: #000;
          width: auto;
          height: auto;
          padding: 0 0 0 0;
          margin: 0 0 0 0;
        }
      </style><div tabindex=\\"0\\" class=\\"hide-tab end\\" aria-hidden=\\"true\\"></div><style>
          .wrapper {
            position: fixed;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            max-width: 100%;
            vertical-align: middle;
            overflow: auto;
            display: flex;
            flex-direction: column;
            box-sizing: border-box;
            background-color: #fff;
            color: #000;
            border-radius: calc(3 * 6px);
            box-shadow: 0 30px 60px rgba(0, 0, 0, 0.12);
            opacity: 0;
            outline: none;
            transform: translate3d(100%, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.6, 0.1, 1);
            font-size: calc(1 * 16px);
            --modal-wrapper-padding-left: calc(1.3125 * 16px);
            --modal-wrapper-padding-right: calc(1.3125 * 16px);
            padding: calc(1.3125 * 16px) var(--modal-wrapper-padding-right)
              calc(1.3125 * 16px) var(--modal-wrapper-padding-left);
            margin: 0 0 0 0;
          }
          .top,
          .bottom {
            width: 100px;
            height: auto;
          }
          .left,
          .right {
            width: 100px;
            height: 100%;
          }
          .top {
            bottom: auto;
            border-top-left-radius: 0;
            border-top-right-radius: 0;
          }
          .left {
            right: auto;
            border-top-left-radius: 0;
            border-bottom-left-radius: 0;
          }
          .bottom {
            top: auto;
            border-bottom-left-radius: 0;
            border-bottom-right-radius: 0;
          }
          .right {
            left: auto;
            border-top-right-radius: 0;
            border-bottom-right-radius: 0;
          }
          .wrapper-enter {
            opacity: 0;
            transform: translate3d(100%, 0, 0);
          }
          .wrapper-enter-active {
            opacity: 1;
            transform: translate3d(0, 0, 0);
          }
          .wrapper-leave {
            opacity: 1;
            transform: translate3d(0, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.2, 0.1, 1);
          }
          .wrapper-leave-active {
            opacity: 0.4;
            transform: translate3d(100%, 0, 0);
          }
          .hide-tab {
            outline: none;
            overflow: hidden;
            width: 0;
            height: 0;
            opacity: 0;
          }
        </style></div>"
`;

exports[`Drawer should render correctly 1`] = `
"<div class=\\"backdrop  backdrop-wrapper-enter\\"><div class=\\"layer\\"></div><div class=\\"position\\"><div class=\\"wrapper right  wrapper-enter\\" role=\\"dialog\\" tabindex=\\"-1\\"><div tabindex=\\"0\\" class=\\"hide-tab start\\" aria-hidden=\\"true\\"></div><h2 class=\\"\\">Drawer</h2><style>
        h2 {
          line-height: 1.6;
          font-weight: normal;
          text-align: center;
          display: inline-flex;
          flex-shrink: 0;
          justify-content: center;
          align-items: center;
          word-break: break-word;
          text-transform: capitalize;
          font-size: calc(1.5 * 16px);
          color: #000;
          width: auto;
          height: auto;
          padding: 0 0 0 0;
          margin: 0 0 0 0;
        }
      </style><p class=\\"\\">This is a drawer</p><style>
        p {
          font-weight: normal;
          display: inline-block;
          text-align: center;
          word-break: break-word;
          text-transform: uppercase;
          color: #666;
          font-size: calc(0.875 * 16px);
          line-height: 1.5em;
          width: auto;
          height: 1.5em;
          padding: 0 0 0 0;
          margin: 0 0 0 0;
        }
      </style><div class=\\"content\\"><p>Some content contained within the drawer.</p></div><style>
        .content {
          position: relative;
          text-align: left;
          font-size: calc(1 * 16px);
          width: auto;
          height: auto;
          padding: calc(1.3125 * 16px) calc(1.3125 * 16px) calc(0.6625 * 16px)
            calc(1.3125 * 16px);
          margin: 0
            calc(var(--modal-wrapper-padding-right) * -1)
            0
            calc(var(--modal-wrapper-padding-left) * -1);
        }

        .content > :global(*:first-child) {
          margin-top: 0;
        }

        .content > :global(*:last-child) {
          margin-bottom: 0;
        }
      </style><div tabindex=\\"0\\" class=\\"hide-tab end\\" aria-hidden=\\"true\\"></div><style>
          .wrapper {
            position: fixed;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            max-width: 100%;
            vertical-align: middle;
            overflow: auto;
            display: flex;
            flex-direction: column;
            box-sizing: border-box;
            background-color: #fff;
            color: #000;
            border-radius: calc(3 * 6px);
            box-shadow: 0 30px 60px rgba(0, 0, 0, 0.12);
            opacity: 0;
            outline: none;
            transform: translate3d(100%, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.6, 0.1, 1);
            font-size: calc(1 * 16px);
            --modal-wrapper-padding-left: calc(1.3125 * 16px);
            --modal-wrapper-padding-right: calc(1.3125 * 16px);
            padding: calc(1.3125 * 16px) var(--modal-wrapper-padding-right)
              calc(1.3125 * 16px) var(--modal-wrapper-padding-left);
            margin: 0 0 0 0;
          }
          .top,
          .bottom {
            width: 100%;
            height: auto;
          }
          .left,
          .right {
            width: auto;
            height: 100%;
          }
          .top {
            bottom: auto;
            border-top-left-radius: 0;
            border-top-right-radius: 0;
          }
          .left {
            right: auto;
            border-top-left-radius: 0;
            border-bottom-left-radius: 0;
          }
          .bottom {
            top: auto;
            border-bottom-left-radius: 0;
            border-bottom-right-radius: 0;
          }
          .right {
            left: auto;
            border-top-right-radius: 0;
            border-bottom-right-radius: 0;
          }
          .wrapper-enter {
            opacity: 0;
            transform: translate3d(100%, 0, 0);
          }
          .wrapper-enter-active {
            opacity: 1;
            transform: translate3d(0, 0, 0);
          }
          .wrapper-leave {
            opacity: 1;
            transform: translate3d(0, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.2, 0.1, 1);
          }
          .wrapper-leave-active {
            opacity: 0.4;
            transform: translate3d(100%, 0, 0);
          }
          .hide-tab {
            outline: none;
            overflow: hidden;
            width: 0;
            height: 0;
            opacity: 0;
          }
        </style></div></div><style>
            .backdrop {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              overflow: auto;
              z-index: 1000;
              -webkit-overflow-scrolling: touch;
              box-sizing: border-box;
              text-align: center;
            }
            .position {
              position: relative;
              z-index: 1001;
              outline: none;
              max-width: 90%;
              width: 100%;
              margin: 20px auto;
              vertical-align: middle;
              display: inline-block;
            }
            .backdrop:before {
              display: inline-block;
              width: 0;
              height: 100%;
              vertical-align: middle;
              content: '';
            }
            .layer {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              width: 100%;
              height: 100%;
              opacity: 0.25;
              background-color: black;
              transition: opacity 0.35s cubic-bezier(0.4, 0, 0.2, 1);
              pointer-events: none;
              z-index: 1000;
            }
            .backdrop-wrapper-enter .layer {
              opacity: 0;
            }
            .backdrop-wrapper-enter-active .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave-active .layer {
              opacity: 0;
            }
          </style></div>"
`;

exports[`Drawer should work correctly with different placement 1`] = `
"<div class=\\"backdrop  backdrop-wrapper-enter\\"><div class=\\"layer\\"></div><div class=\\"position\\"><div class=\\"wrapper top  wrapper-enter\\" role=\\"dialog\\" tabindex=\\"-1\\"><div tabindex=\\"0\\" class=\\"hide-tab start\\" aria-hidden=\\"true\\"></div><p>Some content contained within the drawer.</p><div tabindex=\\"0\\" class=\\"hide-tab end\\" aria-hidden=\\"true\\"></div><style>
          .wrapper {
            position: fixed;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            max-width: 100%;
            vertical-align: middle;
            overflow: auto;
            display: flex;
            flex-direction: column;
            box-sizing: border-box;
            background-color: #fff;
            color: #000;
            border-radius: calc(3 * 6px);
            box-shadow: 0 30px 60px rgba(0, 0, 0, 0.12);
            opacity: 0;
            outline: none;
            transform: translate3d(0, -100%, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.6, 0.1, 1);
            font-size: calc(1 * 16px);
            --modal-wrapper-padding-left: calc(1.3125 * 16px);
            --modal-wrapper-padding-right: calc(1.3125 * 16px);
            padding: calc(1.3125 * 16px) var(--modal-wrapper-padding-right)
              calc(1.3125 * 16px) var(--modal-wrapper-padding-left);
            margin: 0 0 0 0;
          }
          .top,
          .bottom {
            width: 100%;
            height: auto;
          }
          .left,
          .right {
            width: auto;
            height: 100%;
          }
          .top {
            bottom: auto;
            border-top-left-radius: 0;
            border-top-right-radius: 0;
          }
          .left {
            right: auto;
            border-top-left-radius: 0;
            border-bottom-left-radius: 0;
          }
          .bottom {
            top: auto;
            border-bottom-left-radius: 0;
            border-bottom-right-radius: 0;
          }
          .right {
            left: auto;
            border-top-right-radius: 0;
            border-bottom-right-radius: 0;
          }
          .wrapper-enter {
            opacity: 0;
            transform: translate3d(0, -100%, 0);
          }
          .wrapper-enter-active {
            opacity: 1;
            transform: translate3d(0, 0, 0);
          }
          .wrapper-leave {
            opacity: 1;
            transform: translate3d(0, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.2, 0.1, 1);
          }
          .wrapper-leave-active {
            opacity: 0.4;
            transform: translate3d(0, -100%, 0);
          }
          .hide-tab {
            outline: none;
            overflow: hidden;
            width: 0;
            height: 0;
            opacity: 0;
          }
        </style></div></div><style>
            .backdrop {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              overflow: auto;
              z-index: 1000;
              -webkit-overflow-scrolling: touch;
              box-sizing: border-box;
              text-align: center;
            }
            .position {
              position: relative;
              z-index: 1001;
              outline: none;
              max-width: 90%;
              width: 100%;
              margin: 20px auto;
              vertical-align: middle;
              display: inline-block;
            }
            .backdrop:before {
              display: inline-block;
              width: 0;
              height: 100%;
              vertical-align: middle;
              content: '';
            }
            .layer {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              width: 100%;
              height: 100%;
              opacity: 0.25;
              background-color: black;
              transition: opacity 0.35s cubic-bezier(0.4, 0, 0.2, 1);
              pointer-events: none;
              z-index: 1000;
            }
            .backdrop-wrapper-enter .layer {
              opacity: 0;
            }
            .backdrop-wrapper-enter-active .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave-active .layer {
              opacity: 0;
            }
          </style></div>"
`;

exports[`Drawer should work correctly with different placement 2`] = `
"<div class=\\"backdrop  backdrop-wrapper-enter\\"><div class=\\"layer\\"></div><div class=\\"position\\"><div class=\\"wrapper right  wrapper-enter\\" role=\\"dialog\\" tabindex=\\"-1\\"><div tabindex=\\"0\\" class=\\"hide-tab start\\" aria-hidden=\\"true\\"></div><p>Some content contained within the drawer.</p><div tabindex=\\"0\\" class=\\"hide-tab end\\" aria-hidden=\\"true\\"></div><style>
          .wrapper {
            position: fixed;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            max-width: 100%;
            vertical-align: middle;
            overflow: auto;
            display: flex;
            flex-direction: column;
            box-sizing: border-box;
            background-color: #fff;
            color: #000;
            border-radius: calc(3 * 6px);
            box-shadow: 0 30px 60px rgba(0, 0, 0, 0.12);
            opacity: 0;
            outline: none;
            transform: translate3d(100%, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.6, 0.1, 1);
            font-size: calc(1 * 16px);
            --modal-wrapper-padding-left: calc(1.3125 * 16px);
            --modal-wrapper-padding-right: calc(1.3125 * 16px);
            padding: calc(1.3125 * 16px) var(--modal-wrapper-padding-right)
              calc(1.3125 * 16px) var(--modal-wrapper-padding-left);
            margin: 0 0 0 0;
          }
          .top,
          .bottom {
            width: 100%;
            height: auto;
          }
          .left,
          .right {
            width: auto;
            height: 100%;
          }
          .top {
            bottom: auto;
            border-top-left-radius: 0;
            border-top-right-radius: 0;
          }
          .left {
            right: auto;
            border-top-left-radius: 0;
            border-bottom-left-radius: 0;
          }
          .bottom {
            top: auto;
            border-bottom-left-radius: 0;
            border-bottom-right-radius: 0;
          }
          .right {
            left: auto;
            border-top-right-radius: 0;
            border-bottom-right-radius: 0;
          }
          .wrapper-enter {
            opacity: 0;
            transform: translate3d(100%, 0, 0);
          }
          .wrapper-enter-active {
            opacity: 1;
            transform: translate3d(0, 0, 0);
          }
          .wrapper-leave {
            opacity: 1;
            transform: translate3d(0, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.2, 0.1, 1);
          }
          .wrapper-leave-active {
            opacity: 0.4;
            transform: translate3d(100%, 0, 0);
          }
          .hide-tab {
            outline: none;
            overflow: hidden;
            width: 0;
            height: 0;
            opacity: 0;
          }
        </style></div></div><style>
            .backdrop {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              overflow: auto;
              z-index: 1000;
              -webkit-overflow-scrolling: touch;
              box-sizing: border-box;
              text-align: center;
            }
            .position {
              position: relative;
              z-index: 1001;
              outline: none;
              max-width: 90%;
              width: 100%;
              margin: 20px auto;
              vertical-align: middle;
              display: inline-block;
            }
            .backdrop:before {
              display: inline-block;
              width: 0;
              height: 100%;
              vertical-align: middle;
              content: '';
            }
            .layer {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              width: 100%;
              height: 100%;
              opacity: 0.25;
              background-color: black;
              transition: opacity 0.35s cubic-bezier(0.4, 0, 0.2, 1);
              pointer-events: none;
              z-index: 1000;
            }
            .backdrop-wrapper-enter .layer {
              opacity: 0;
            }
            .backdrop-wrapper-enter-active .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave-active .layer {
              opacity: 0;
            }
          </style></div>"
`;

exports[`Drawer should work correctly with different placement 3`] = `
"<div class=\\"backdrop  backdrop-wrapper-enter\\"><div class=\\"layer\\"></div><div class=\\"position\\"><div class=\\"wrapper bottom  wrapper-enter\\" role=\\"dialog\\" tabindex=\\"-1\\"><div tabindex=\\"0\\" class=\\"hide-tab start\\" aria-hidden=\\"true\\"></div><p>Some content contained within the drawer.</p><div tabindex=\\"0\\" class=\\"hide-tab end\\" aria-hidden=\\"true\\"></div><style>
          .wrapper {
            position: fixed;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            max-width: 100%;
            vertical-align: middle;
            overflow: auto;
            display: flex;
            flex-direction: column;
            box-sizing: border-box;
            background-color: #fff;
            color: #000;
            border-radius: calc(3 * 6px);
            box-shadow: 0 30px 60px rgba(0, 0, 0, 0.12);
            opacity: 0;
            outline: none;
            transform: translate3d(0, 100%, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.6, 0.1, 1);
            font-size: calc(1 * 16px);
            --modal-wrapper-padding-left: calc(1.3125 * 16px);
            --modal-wrapper-padding-right: calc(1.3125 * 16px);
            padding: calc(1.3125 * 16px) var(--modal-wrapper-padding-right)
              calc(1.3125 * 16px) var(--modal-wrapper-padding-left);
            margin: 0 0 0 0;
          }
          .top,
          .bottom {
            width: 100%;
            height: auto;
          }
          .left,
          .right {
            width: auto;
            height: 100%;
          }
          .top {
            bottom: auto;
            border-top-left-radius: 0;
            border-top-right-radius: 0;
          }
          .left {
            right: auto;
            border-top-left-radius: 0;
            border-bottom-left-radius: 0;
          }
          .bottom {
            top: auto;
            border-bottom-left-radius: 0;
            border-bottom-right-radius: 0;
          }
          .right {
            left: auto;
            border-top-right-radius: 0;
            border-bottom-right-radius: 0;
          }
          .wrapper-enter {
            opacity: 0;
            transform: translate3d(0, 100%, 0);
          }
          .wrapper-enter-active {
            opacity: 1;
            transform: translate3d(0, 0, 0);
          }
          .wrapper-leave {
            opacity: 1;
            transform: translate3d(0, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.2, 0.1, 1);
          }
          .wrapper-leave-active {
            opacity: 0.4;
            transform: translate3d(0, 100%, 0);
          }
          .hide-tab {
            outline: none;
            overflow: hidden;
            width: 0;
            height: 0;
            opacity: 0;
          }
        </style></div></div><style>
            .backdrop {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              overflow: auto;
              z-index: 1000;
              -webkit-overflow-scrolling: touch;
              box-sizing: border-box;
              text-align: center;
            }
            .position {
              position: relative;
              z-index: 1001;
              outline: none;
              max-width: 90%;
              width: 100%;
              margin: 20px auto;
              vertical-align: middle;
              display: inline-block;
            }
            .backdrop:before {
              display: inline-block;
              width: 0;
              height: 100%;
              vertical-align: middle;
              content: '';
            }
            .layer {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              width: 100%;
              height: 100%;
              opacity: 0.25;
              background-color: black;
              transition: opacity 0.35s cubic-bezier(0.4, 0, 0.2, 1);
              pointer-events: none;
              z-index: 1000;
            }
            .backdrop-wrapper-enter .layer {
              opacity: 0;
            }
            .backdrop-wrapper-enter-active .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave-active .layer {
              opacity: 0;
            }
          </style></div>"
`;

exports[`Drawer should work correctly with different placement 4`] = `
"<div class=\\"backdrop  backdrop-wrapper-enter\\"><div class=\\"layer\\"></div><div class=\\"position\\"><div class=\\"wrapper left  wrapper-enter\\" role=\\"dialog\\" tabindex=\\"-1\\"><div tabindex=\\"0\\" class=\\"hide-tab start\\" aria-hidden=\\"true\\"></div><p>Some content contained within the drawer.</p><div tabindex=\\"0\\" class=\\"hide-tab end\\" aria-hidden=\\"true\\"></div><style>
          .wrapper {
            position: fixed;
            top: 0;
            left: 0;
            right: 0;
            bottom: 0;
            max-width: 100%;
            vertical-align: middle;
            overflow: auto;
            display: flex;
            flex-direction: column;
            box-sizing: border-box;
            background-color: #fff;
            color: #000;
            border-radius: calc(3 * 6px);
            box-shadow: 0 30px 60px rgba(0, 0, 0, 0.12);
            opacity: 0;
            outline: none;
            transform: translate3d(-100%, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.6, 0.1, 1);
            font-size: calc(1 * 16px);
            --modal-wrapper-padding-left: calc(1.3125 * 16px);
            --modal-wrapper-padding-right: calc(1.3125 * 16px);
            padding: calc(1.3125 * 16px) var(--modal-wrapper-padding-right)
              calc(1.3125 * 16px) var(--modal-wrapper-padding-left);
            margin: 0 0 0 0;
          }
          .top,
          .bottom {
            width: 100%;
            height: auto;
          }
          .left,
          .right {
            width: auto;
            height: 100%;
          }
          .top {
            bottom: auto;
            border-top-left-radius: 0;
            border-top-right-radius: 0;
          }
          .left {
            right: auto;
            border-top-left-radius: 0;
            border-bottom-left-radius: 0;
          }
          .bottom {
            top: auto;
            border-bottom-left-radius: 0;
            border-bottom-right-radius: 0;
          }
          .right {
            left: auto;
            border-top-right-radius: 0;
            border-bottom-right-radius: 0;
          }
          .wrapper-enter {
            opacity: 0;
            transform: translate3d(-100%, 0, 0);
          }
          .wrapper-enter-active {
            opacity: 1;
            transform: translate3d(0, 0, 0);
          }
          .wrapper-leave {
            opacity: 1;
            transform: translate3d(0, 0, 0);
            transition: opacity, transform 400ms cubic-bezier(0.1, 0.2, 0.1, 1);
          }
          .wrapper-leave-active {
            opacity: 0.4;
            transform: translate3d(-100%, 0, 0);
          }
          .hide-tab {
            outline: none;
            overflow: hidden;
            width: 0;
            height: 0;
            opacity: 0;
          }
        </style></div></div><style>
            .backdrop {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              overflow: auto;
              z-index: 1000;
              -webkit-overflow-scrolling: touch;
              box-sizing: border-box;
              text-align: center;
            }
            .position {
              position: relative;
              z-index: 1001;
              outline: none;
              max-width: 90%;
              width: 100%;
              margin: 20px auto;
              vertical-align: middle;
              display: inline-block;
            }
            .backdrop:before {
              display: inline-block;
              width: 0;
              height: 100%;
              vertical-align: middle;
              content: '';
            }
            .layer {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              width: 100%;
              height: 100%;
              opacity: 0.25;
              background-color: black;
              transition: opacity 0.35s cubic-bezier(0.4, 0, 0.2, 1);
              pointer-events: none;
              z-index: 1000;
            }
            .backdrop-wrapper-enter .layer {
              opacity: 0;
            }
            .backdrop-wrapper-enter-active .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave .layer {
              opacity: 0.25;
            }
            .backdrop-wrapper-leave-active .layer {
              opacity: 0;
            }
          </style></div>"
`;
